src/main/kotlin/com/manimdsl/stylesheet/Stylesheet.kt

```diff
@@ -97,10 +97,10 @@ class Stylesheet(private val stylesheetPath: String?, private val symbolTableVis
         val dataStructureStyle =
             stylesheet.dataStructures.getOrDefault(value.toString(), StyleProperties())
         val style = stylesheet.variables.getOrDefault(identifier, dataStructureStyle)
-
         val animationStyle = (style.animate ?: AnimationProperties()) merge (dataStructureStyle.animate ?: AnimationProperties())
-        // Return null if there is no style to make sure null checks work throughout executor
-        return if (animationStyle == AnimationProperties()) null else animationStyle
+
+        // Returns null if there is no style to make sure null checks work throughout executor
+        return if (animationStyle.borderColor == null && animationStyle.textColor == null) null else animationStyle
```

Comment:
How is this different to checking just the AnimationProperties() class?

Reply:
Is it something to do with the pointer property? I also faced this issue and fixed it some other way, not sure which one is right though

Reply:
It might be, I'm not sure. The check doesn't work as expected when you check the `AnimationProperties()` class

Reply:
I fixed it like this: https://github.com/ManimDSL/ManimDSLCompiler/pull/66/commits/c37d55cadf6e4c35653eb0f8e47021dbe3c9c23f

Reply:
The problem was it was checking if the whole class was null as a pose to the specific property I think

Reply:
Yeah this is the same issue I was trying to fix actually

Reply:
I think some parts of the code rely on this function returning null sometimes, in order to figure out whether something should be animated. Currently the check on `AnimationProperties` always fails

Reply:
Hmm I'm just thinking that this shouldn't happen, equality on data classes is comprised of all the values inside it, so checking each value for null and checking the whole thing (with each internal value as null) should be exactly the same thing

Reply:
Yeah I was really confused by it

Reply:
Was there another case other than the one in https://github.com/ManimDSL/ManimDSLCompiler/commit/c37d55cadf6e4c35653eb0f8e47021dbe3c9c23f ? 

Because we should look into the usages of animated styles instead of fixing it at this point I think?

Reply:
We should never check on the whole property only, it should be on each individual property too, which I think we might do, I'll have a look

Reply:
[Here](https://github.com/ManimDSL/ManimDSLCompiler/blob/55bcdf4b2eddd20f7ec75ee91765e2ffed96d505/src/main/kotlin/com/manimdsl/runtime/ASTExecutor.kt#L455)